### PR TITLE
Consume vm fdt

### PR DESCRIPTION
**Pull request by AlexandruCihodaru**
# Reason for This PR

Consume rust-vmm/vm-fdt

## Description of Changes

-> removed libfdt-bindings
-> Replaced libfdt tool with vm-fdt from rust-vmm.
-> Deleted unused helper functions.

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any API changes are reflected in `firecracker/swagger.yaml`.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

### CODE REVIEW

**Review by serban300 — COMMENTED** [review]
The PR looks good. I just have a couple of nits on the code and a high level one: I would change the order of the commits in order to make each one compile. Something like:
1. Include vm-fdt crate
2. Use vm-fdt
3. Delete libfdt-bindings 

Also I was curious. Does the size of the Firecracker binary decreases with this change ? If so can we also update `test_binary_size.py` ?

File: src/arch/Cargo.toml [hunk 1]
```diff
@@ -11,13 +11,11 @@ libc = ">=0.2.39"
...
 
 arch_gen = { path = "../arch_gen" }
 logger = { path = "../logger" }
 utils = { path = "../utils" }
 
 [dev-dependencies]
-device_tree = ">=1.1.0"
-
-[target.'cfg(target_arch="aarch64")'.dependencies]
-libfdt-bindings = { path = "../libfdt-bindings" }
```
Comment (serban300) [inline src/arch/Cargo.toml:23]: Nit: This should be part of the 3rd commit. Also should we add a new line at the end of the file ?

**Review by serban300 — DISMISSED** [review]

**Review by dianpopa — DISMISSED** [review]
I would like to also review this PR so blocking it until I get a chance to do so

File: src/arch/src/aarch64/fdt.rs [hunk 2]
```diff
@@ -746,7 +547,7 @@ mod tests {
         buf.extend_from_slice(saved_dtb_bytes);
 
         set_size(&mut buf, pos, val);
-        set_size(&mut current_dtb_bytes, pos, val);
+        println!("{} {}", buf.len(), current_dtb_bytes.len());
```
Comment (dianpopa) [inline src/arch/src/aarch64/fdt.rs:550]: leftover println?

File: src/arch/Cargo.toml [hunk 3]
```diff
@@ -18,7 +18,4 @@ logger = { path = "../logger" }
...
\ No newline at end of file
+device_tree = ">=1.1.0"
```
Comment (dianpopa) [inline src/arch/Cargo.toml:21]: nit: newline

File: src/arch/src/aarch64/fdt.rs [hunk 4]
```diff
@@ -5,24 +5,20 @@
...
 use super::super::InitrdConfig;
 use super::cache_info::{read_cache_config, CacheEntry};
 use super::get_fdt_addr;
 use super::gic::GICDevice;
-use super::layout::FDT_MAX_SIZE;
-use crate::aarch64::fdt::Error::CstringFDTTransform;
 use vm_memory::{Address, Bytes, GuestAddress, GuestMemory, GuestMemoryError, GuestMemoryMmap};
 
+extern crate vm_fdt;
+use self::vm_fdt::{Error as VmFdtError, FdtWriter, FdtWriterNode};
```
Comment (dianpopa) [inline src/arch/src/aarch64/fdt.rs:20]: `extern crate` is not needed.
it suffices:
`use vm_fdt::...`

**Review by serban300 — APPROVED** [review]

**Review by dianpopa — APPROVED** [review]

**Pull request merged**